src/main/java/org/jitsi/srtp/SrtpCryptoContext.java

```diff
@@ -330,12 +330,130 @@ private long guessIndex(int seqNo)
         return (((long) guessedROC) << 16) | seqNo;
     }
 
+    /**
+     * Determine if this packet should be processed with "cryptex"
+     * (header extension encryption)
+     */
+    private boolean useCryptex(ByteArrayBuffer pkt, boolean encrypting)
+    {
+        if (!SrtpPacketUtils.getExtensionBit(pkt))
+        {
+            return false;
+        }
+
+        int type = SrtpPacketUtils.getExtensionType(pkt);
+        if (encrypting)
+        {
+            if (!policy.isCryptexEnabled())
+            {
+                return false;
+            }
+
+            switch (type)
+            {
+            case 0xBEDE:
+            case 0x1000:
+                return true;
+            default:
+                return false;
+            }
+        }
+        else
+        {
+            switch (type)
+            {
+            case 0xC0DE:
+            case 0xC2DE:
+                return true;
+            default:
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Given an input header type, return the value as transformed
+     * by cryptex processing.  Assumes useCryptex() returns true for this packet.
+     */
+    private int getTransformedHeaderType(int type)
+    {
+        switch (type)
+        {
+        case 0xBEDE:
+            return 0xC0DE;
+        case 0x1000:
+            return 0xC2DE;
+        case 0xC0DE:
+            return 0xBEDE;
+        case 0xC2DE:
+            return 0x1000;
+        default:
+            /* Can't happen if useCryptex returned true */
+            throw new IllegalStateException(String.format("Invalid header type 0x%4X", type));
+        }
+    }
+
+    /**
+     * Transform a packet's "defined by profile" header extension field for cryptex.
+     * This should be a packet for which useCryptex() has returned trye.
+     */
+    private void transformCryptexType(ByteArrayBuffer pkt)
+    {
+        int type = SrtpPacketUtils.getExtensionType(pkt);
+        int newType = getTransformedHeaderType(type);
+        SrtpPacketUtils.setExtensionType(pkt, newType);
+    }
+
+    /**
+     * Pre-process a packet so it is ready for cryptex encryption.
```

Comment:
Isn't this (and `cryptexPostprocess` below) used for decryption as well? Or maybe I interpret "encryption" too narrowly